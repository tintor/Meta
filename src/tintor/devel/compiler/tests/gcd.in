void main
	out gdc(96, 1024)

int gcd(int a, int b)
	if a >= b
		if b > 0
			gcd a%b, a
		else
			a
	else
		gdc b, a