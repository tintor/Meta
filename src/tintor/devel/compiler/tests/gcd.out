(class-body
	(func main (args)
		(code
			(call print (call gcd (int 96) (int 1024)))			
		)
	)
	(func gdc (args a b)
		(code
			(if
				(>= (name a) (name b))
				(if
					(> (name b) (int 0))
					(call gcd (% (call a) (call b)) (call a))
					(call a)
				)
				(call gcd (call b) (call a))
			)
		)
	)
)

main
	print gdc(96, 1024)

gcd(a, b)
	if a >= b
		if b > 0
			gcd a%b, a
		else
			a
	else
		gdc b, a

class-body := func+
func := name